### PR TITLE
[ci, vllm] chore: update vllm-omni 0.18.0 official release and Miscellaneous

### PR DESCRIPTION
**Pull request by AndyZhou952**
### What does this PR do?

> Add **concise** overview of what this PR aims to achieve or accomplish. Reference related GitHub issues and PRs that help with the review.

- **vLLM / vllm-omni 0.18.0** official release in the vLLM-Omni CI workflow (replacing a git SHA install), adds TP support. 
- Aligns the FlowGRPO `QwenImagePipelineWithLogProb` example with upstream’s test pipeline `__init__` pattern [`in vllm-omni`](https://github.com/vllm-project/vllm-omni/blob/v0.18.0/tests/e2e/offline_inference/custom_pipeline/qwen_image_pipeline_with_logprob.py). 
- Updates Omni sampling tests to use `true_cfg_scale` instead of `guidance_scale` for Qwen-Image-style CFG. 
- Enables `tensor_model_parallel_size = 2` in the diffusion agent loop test. **Remark: `tiny-random/Qwen-Image` has config `num_attention_heads` being 1, and thus we need to manually create a tmp file based on it to properly test TP behavior.**

### Checklist Before Starting

- [x] Search for similar PRs. Paste at least one query link here: ...
- [x] Format the PR title as `[{modules}] {type}: {description}` (This will be checked by the CI)
  - `{modules}` include `fsdp`, `megatron`, `veomni`, `sglang`, `vllm`, `rollout`, `trainer`, `ci`, `training_utils`, `recipe`, `hardware`, `deployment`, `ray`, `worker`, `single_controller`, `misc`, `perf`, `model`, `algo`, `env`, `tool`, `ckpt`, `doc`, `data`, `cfg`, `reward`, `fully_async`, `one_step_off`
  - If this PR involves multiple modules, separate them with `,` like `[megatron, fsdp, doc]`
  - `{type}` is in `feat`, `fix`, `refactor`, `chore`, `test`
  - If this PR breaks any API (CLI arguments, config, function signature, etc.), add `[BREAKING]` to the beginning of the title.
  - Example: `[BREAKING][fsdp, megatron] feat: dynamic batching`

### Test

`tests/experimental/agent_loop/test_diffusion_agent_loop.py` and `test_vllm_omni_generate.py` updated and passed
```
tests/experimental/agent_loop/test_diffusion_agent_loop.py .                                          [100%]

======================================== warnings summary =========================================
../../miniforge3/envs/vllm-omni-dev/lib/python3.12/site-packages/requests/__init__.py:113
  /scratch/fq9hpsac/mikecheung/miniforge3/envs/vllm-omni-dev/lib/python3.12/site-packages/requests/__init__.py:113: RequestsDependencyWarning: urllib3 (2.6.3) or chardet (6.0.0.post1)/charset_normalizer (3.4.4) doesn't match a supported version!
    warnings.warn(

../../miniforge3/envs/vllm-omni-dev/lib/python3.12/site-packages/ray/util/state/util.py:55
  /scratch/fq9hpsac/mikecheung/miniforge3/envs/vllm-omni-dev/lib/python3.12/site-packages/ray/util/state/util.py:55: DeprecationWarning: Ray state API is no longer experimental. Please import from `ray.util.state` instead. Importing from `ray.experimental` will be deprecated in future releases.
    warnings.warn(

<frozen importlib._bootstrap>:488
  <frozen importlib._bootstrap>:488: DeprecationWarning: builtin type SwigPyPacked has no __module__ attribute

<frozen importlib._bootstrap>:488
  <frozen importlib._bootstrap>:488: DeprecationWarning: builtin type SwigPyObject has no __module__ attribute

../../miniforge3/envs/vllm-omni-dev/lib/python3.12/site-packages/torch/jit/_script.py:362: 14 warnings
  /scratch/fq9hpsac/mikecheung/miniforge3/envs/vllm-omni-dev/lib/python3.12/site-packages/torch/jit/_script.py:362: DeprecationWarning: `torch.jit.script_method` is deprecated. Please switch to `torch.compile` or `torch.export`.
    warnings.warn(

tests/experimental/agent_loop/test_diffusion_agent_loop.py::test_single_turn
  /scratch/fq9hpsac/mikecheung/gitlocal/verl/tests/experimental/agent_loop/test_diffusion_agent_loop.py:63: UserWarning: 
  The version_base parameter is not specified. 
  Please specify a compatability version level, or None. 
  Will assume defaults for version 1.1
    with initialize_config_dir(config_dir=os.path.abspath("verl/trainer/config")):

tests/experimental/agent_loop/test_diffusion_agent_loop.py::test_single_turn
  /scratch/fq9hpsac/mikecheung/miniforge3/envs/vllm-omni-dev/lib/python3.12/site-packages/ray/_private/worker.py:2052: FutureWarning: Tip: In future versions of Ray, Ray will no longer override accelerator visible devices env var if num_gpus=0 or num_gpus=None (default). To enable this behavior and turn off this error message, set RAY_ACCEL_ENV_VAR_OVERRIDE_ON_ZERO=0
    warnings.warn(

tests/experimental/agent_loop/test_diffusion_agent_loop.py::test_single_turn
  /scratch/fq9hpsac/mikecheung/miniforge3/envs/vllm-omni-dev/lib/python3.12/site-packages/pydub/utils.py:14: DeprecationWarning: 'audioop' is deprecated and slated for removal in Python 3.13
    import audioop

tests/experimental/agent_loop/test_diffusion_agent_loop.py::test_single_turn
  /scratch/fq9hpsac/mikecheung/miniforge3/envs/vllm-omni-dev/lib/python3.12/site-packages/pydub/utils.py:170: RuntimeWarning: Couldn't find ffmpeg or avconv - defaulting to ffmpeg, but may not work
    warn("Couldn't find ffmpeg or avconv - defaulting to ffmpeg, but may not work", RuntimeWarning)

tests/experimental/agent_loop/test_diffusion_agent_loop.py::test_single_turn
  /scratch/fq9hpsac/mikecheung/miniforge3/envs/vllm-omni-dev/lib/python3.12/site-packages/vllm_omni/entrypoints/openai/protocol/audio.py:112: PydanticDeprecatedSince20: Support for class-based `config` is deprecated, use ConfigDict instead. Deprecated in Pydantic V2.0 to be removed in V3.0. See Pydantic V2 Migration Guide at https://errors.pydantic.dev/2.12/migration/
    class CreateAudio(BaseModel):

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
========================= 1 passed, 23 warnings in 105.69s (0:01:45) ==========================
sys:1: DeprecationWarning: builtin type swigvarlink has no __module__ attribute
```

> For changes that can not be tested by CI (e.g., algorithm implementation, new model support), validate by experiment(s) and show results like training curve plots, evaluation results, etc.

### API and Usage Example

NA

> Demonstrate how the API changes if any, and provide usage example(s) if possible.

```python
# Add code snippet or script demonstrating how to use this
```

### Design & Code Changes

> Demonstrate the high-level design if this PR is complex, and list the specific changes.
NA

### Checklist Before Submitting

> [!IMPORTANT]
> Please check all the following items before requesting a review, otherwise the reviewer might deprioritize this PR for review.

- [x] Read the [Contribute Guide](https://github.com/volcengine/verl/blob/main/CONTRIBUTING.md).
- [x] Apply [pre-commit checks](https://github.com/volcengine/verl/blob/main/CONTRIBUTING.md#code-linting-and-formatting): `pre-commit install && pre-commit run --all-files --show-diff-on-failure --color=always`
- [x] Add / Update [the documentation](https://github.com/volcengine/verl/tree/main/docs).
- [x] Add unit or end-to-end test(s) to [the CI workflow](https://github.com/volcengine/verl/tree/main/.github/workflows) to cover all the code. If not feasible, explain why: ...
- [x] Once your PR is ready for CI, send a message in [the `ci-request` channel](https://verl-project.slack.com/archives/C091TCESWB1) in [the `verl` Slack workspace](https://join.slack.com/t/verl-project/shared_invite/zt-3855yhg8g-CTkqXu~hKojPCmo7k_yXTQ). (If not accessible, please try [the Feishu group (飞书群)](https://applink.larkoffice.com/client/chat/chatter/add_by_link?link_token=772jd4f1-cd91-441e-a820-498c6614126a).)
- [x] If your PR is related to the `recipe` submodule, please also update the reference to the submodule commit via `git submodule update --remote` or `cd recipe && git pull origin main`.

### CODE REVIEW

**Comment by AndyZhou952** [comment]
> Looks clear to me. Can you attach the test results as well?

Sure. Just updated the PR description w/ test log for `test_diffusion_agent_loop.py` w/ updated tp=2

File: tests/experimental/agent_loop/test_diffusion_agent_loop.py [hunk 1]
```diff
@@ -24,14 +26,45 @@
...
+    """Copy base model and recreate transformer on-the-fly with TP-compatible head count.
+
+    The tiny-random Qwen-Image model has num_attention_heads=1 in its transformer config,
+    which is not divisible by tensor_model_parallel_size=2. This helper copies the full
+    model directory (vae, text_encoder, tokenizer, scheduler) and overwrites only the
+    transformer component with a freshly-initialized one that has the desired head count.
+    """
+    from diffusers import QwenImageTransformer2DModel
+
+    tmp_dir = tempfile.mkdtemp()
```
Comment (zhtmike) [inline tests/experimental/agent_loop/test_diffusion_agent_loop.py]: should we use a `with xxx` context manager for temp dir, to avoid space keep increasing in CI. Since the temp model is not small.

File: .github/workflows/vllm_omni.yml [hunk 2]
```diff
@@ -111,7 +111,8 @@ jobs:
           pip3 install --no-deps -e .
       - name: Install vllm-omni
         run: |
-          pip3 install git+https://github.com/vllm-project/vllm-omni.git@3158e9d28f1286119756604e9c40292cee4808d3
+          pip3 install 'vllm==0.18.0'
```
Comment (wuxibin89) [inline .github/workflows/vllm_omni.yml]: Please do not install vllm in ci, we're upgrading vllm ci image: https://github.com/verl-project/verl/pull/5724

Comment (AndyZhou952) [reply .github/workflows/vllm_omni.yml]: thanks & removed the vllm part in the CI

File: tests/experimental/agent_loop/test_diffusion_agent_loop.py (hunk 1 above)
Comment (AndyZhou952) [reply tests/experimental/agent_loop/test_diffusion_agent_loop.py]: updated to use context manager - agree that it will be more robust :)
